### PR TITLE
Added a bunch of unit tests

### PR DESCRIPTION
**Pull request by VillanuevaK**
Also sped up retrieval times of functions using hashmaps.

### CODE REVIEW

File: src/main/java/codeu/model/data/Event.java [hunk 1]
```diff
@@ -84,16 +90,66 @@ public Event(Message message) {
...
     messageCreationTime = message.getCreationTime();
     authorIdForMessage = message.getAuthorId();
-    conversationTitleOfMessage = conversationStore.getConversationWithId(message.getConversationId()).getTitle(); //null error is here
+    try {
+      conversationStore.getConversationWithId(message
+        .getConversationId()).getTitle();
+    }
+    catch (NullPointerException e) {
+      System.err.println("_______________________________________________"  +
+                         "\n CAUGHT EXCEPTION_____________________________" +
```
Comment (yufangqq) [inline src/main/java/codeu/model/data/Event.java:103]: nit:
You can keep this. But generally, unit test is to serve the app code. Normally do NOT put any logic/message specific for unit test in the application code.
For example, let say the app is running for some reason this NullPointerException is trigger, then the user will see a message about unit test, that is weird.

File: src/main/java/codeu/model/store/basic/ConversationStore.java [hunk 2]
```diff
@@ -58,11 +59,15 @@ public static ConversationStore getTestInstance(PersistentStorageAgent persisten
 
   /** The in-memory list of Conversations. */
   private List<Conversation> conversations;
+  private HashMap<UUID, Conversation> conversationMap_IdKeyValues;
```
Comment (yufangqq) [inline src/main/java/codeu/model/store/basic/ConversationStore.java]: 1) If you want to keep the variable name, you should still to the lowerCamelCase naming convention e.g., conversationmapIdKeyValues.

2) But I would suggest simply DROP the '_IdKeyValues'. This is easily inferred by the fact that the variable is a HashMap. Sometime we do NOT even say Map (i.e., we normally do NOT put the type of the variable inside the variable names). But you already have a 'conversations' variable, so 'conversationMap' to distinguish from that is okay.

File: src/main/java/codeu/model/store/basic/ConversationStore.java [hunk 3]
```diff
@@ -58,11 +59,15 @@ public static ConversationStore getTestInstance(PersistentStorageAgent persisten
...
+  private HashMap<String, Conversation> conversationMap_StringKeyValues;
```
Comment (yufangqq) [inline src/main/java/codeu/model/store/basic/ConversationStore.java]: Suggest

private HashMap<UUID, Conversation> conversationIdMap;
private HashMap<String, Conversation> conversationTitleMap;

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 4]
```diff
@@ -59,11 +61,15 @@ public static UserStore getTestInstance(PersistentStorageAgent persistentStorage
 
   /** The in-memory list of Users. */
   private List<User> users;
+  private HashMap<UUID, User> userMap_IdKeyValues;
```
Comment (yufangqq) [inline src/main/java/codeu/model/store/basic/UserStore.java]: ditto about variable naming.

File: src/test/java/codeu/controller/ActivityServletTest.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    /* Allows us to use EventStore for all tests. */
+    activityServlet.setEventStore(mockEventStore);
+  }
+
+  /**
+   * Test doGet() in the ActivityServlet to make sure it correctly 
+   * passes the list of event to the activity.jsp file.
+   */
+  @Test
+  public void testDoGet() throws IOException, ServletException {
```
Comment (yufangqq) [inline src/test/java/codeu/controller/ActivityServletTest.java:62]: Should there be also a test for testDoPost(), maybe add a TODO for that test?

**Review by yufangqq — COMMENTED** [review]
Overall I am very pleased with all the tests you have added. They demonstrated good understanding of how these unit test / frameworks work.

@lucieleblanc @laurenfraser2 @VillanuevaK 
One good practice about "unit test" is that we should try our best to minimize the amount change to make to the "app code" in order to support the "unit test". In other words, we should NOT change the "app code" a lot just so we can do unit test. We should try our best to test the existing interface of the "app code". Another way to think about is that the "app code" should not be aware of the unit test code / how it is being tested.

File: src/main/java/codeu/model/data/Event.java (hunk 1 above)
Comment (VillanuevaK) [reply src/main/java/codeu/model/data/Event.java:103]: Haha got it.

**Pull request merged**